### PR TITLE
improve datachain subtract

### LINKED ISSUE #181: DataChain.subtract() on differently named columns
We need to handle use cases where signal names do not match. Like: "file.name" == "wds.laion.file.name" AND "file.etag" == "wds.laion.file.etag". That probably means adding a `right_on` argument to `subtract()`, similarly to `merge()`.
Follow-up to #103 and #157.

### PR DESCRIPTION
**Pull request by EdwardLi-coder**
Fix: DataChain.subtract() on differently named columns 

Fixes #181

### CODE REVIEW

File: src/datachain/query/dataset.py [hunk 1]
```diff
@@ -296,15 +296,23 @@ def q(*columns):
 
 @frozen
 class Subtract(DatasetDiffOperation):
-    on: Sequence[str]
+    on: Sequence[Union[str, tuple[str, str]]]
```
Comment (rlamy) [inline src/datachain/query/dataset.py:299]: Shouldn't this be `Sequence[tuple[str, str]]`?

Comment (EdwardLi-coder) [reply src/datachain/query/dataset.py:299]: Yeah.I will update it.Thank you.

Comment (EdwardLi-coder) [reply src/datachain/query/dataset.py:299]: > Shouldn't this be `Sequence[tuple[str, str]]`?

Yeah.I will update it.Thank you.

File: tests/unit/lib/test_datachain.py [hunk 2]
```diff
@@ -1811,3 +1846,32 @@ def test_from_csv_nan_inf(tmp_dir, test_session):
     assert np.isnan(res[0])
     assert np.isposinf(res[1])
     assert np.isneginf(res[2])
+
+
+def test_subtract_with_different_column_names(test_session):
```
[the remark this answers is not in the document]
Comment (EdwardLi-coder) [reply tests/unit/lib/test_datachain.py:1851]: > Is this test useful? It seems to duplicate the addition to `test_subtract()`.

Sorry.I forgot to delete it and will do so now

**Pull request merged**